elide-core/src/main/java/com/yahoo/elide/core/filter/dialect/RSQLFilterDialect.java

```diff
@@ -0,0 +1,315 @@
+    public RSQLFilterDialect(EntityDictionary dictionary) {
+        this(dictionary, RSQLOperators.defaultOperators());
+    }
+
+
+    @Override
+    public FilterExpression parseGlobalExpression(String path, MultivaluedMap<String, String> queryParams)
+            throws ParseException {
+        if (queryParams.size() != 1) {
+            throw new ParseException("There can only be a single filter query parameter");
+        }
+
+        MultivaluedMap.Entry<String, List<String>> entry = queryParams.entrySet().iterator().next();
+        String queryParamName = entry.getKey();
+
+        if (!queryParamName.equals("filter")) {
+            throw new ParseException("Invalid query parameter: " + queryParamName);
+        }
+        List<String> queryParamValues = entry.getValue();
+
+        if (queryParamValues.size() != 1) {
+            throw new ParseException("There can only be a single filter query parameter");
+        }
+
+        String queryParamValue = queryParamValues.get(0);
+
+        /* Extract the last collection in the URL */
+        path = Paths.get(path).normalize().toString().replace(File.separatorChar, '/');
+        if (path.startsWith("/")) {
+            path = path.substring(1);
+        }
+
+        String[] pathComponents = path.split("/");
+        String lastPathComponent = "";
+        if (pathComponents.length > 0) {
+            lastPathComponent = pathComponents[pathComponents.length - 1];
+        }
+
+        Class entityType = dictionary.getEntityClass(lastPathComponent);
+        if (entityType == null) {
+            throw new ParseException("No such collection: " + lastPathComponent);
+        }
+
+        try {
+            Node ast = parser.parse(queryParamValue);
+            RSQL2ExpressionFilterVisitor visitor = new RSQL2ExpressionFilterVisitor(true);
+            FilterExpression filterExpression = ast.accept(visitor, entityType);
+            return filterExpression;
+        } catch (RSQLParserException e) {
+            throw new ParseException(e.getMessage());
+        }
+    }
+
+    @Override
+    public Map<String, FilterExpression> parseTypedExpression(String path, MultivaluedMap<String, String>
+            queryParams) throws ParseException {
+
+        Map<String, FilterExpression> expressionByType = new HashMap<>();
+
+        for (MultivaluedMap.Entry<String, List<String>> entry : queryParams.entrySet()) {
+
+            String paramName = entry.getKey();
+            List<String> paramValues = entry.getValue();
+
+            Matcher matcher = typedFilterPattern.matcher(paramName);
+            if (matcher.find()) {
+                String typeName = matcher.group(1);
+                if (paramValues.size() != 1) {
+                    throw new ParseException("Exactly one RSQL expression must be defined for type : " + typeName);
+                }
+
+                String expressionText = paramValues.get(0);
+
+                Class entityType = dictionary.getEntityClass(typeName);
```

Comment:
How would this work for a filter like: `filter[book.author.id]`. I'm guessing that in `dictionary.getEntityClass('book.author.id')` is going to break stuff